src/Symfony/Component/Polyfill/Normalizer.php

```diff
@@ -0,0 +1,292 @@
+
+    public static function isNormalized($s, $form = self::NFC)
+    {
+        if (strspn($s .= '', self::$ASCII) === strlen($s)) {
+            return true;
+        }
+        if (self::NFC === $form && preg_match('//u', $s) && !preg_match('/[^\x00-\x{2FF}]/u', $s)) {
+            return true;
+        }
+
+        return false; // Pretend false as quick checks implementented in PHP won't be so quick
+    }
+
+    public static function normalize($s, $form = self::NFC)
+    {
+        if (!preg_match('//u', $s .= '')) {
+            return false;
+        }
+
+        switch ($form) {
+            case self::NONE: return $s;
+            case self::NFC: $C = true; $K = false; break;
+            case self::NFD: $C = false; $K = false; break;
+            case self::NFKC: $C = true; $K = true; break;
+            case self::NFKD: $C = false; $K = true; break;
+            default: return false;
+        }
+
+        if ('' === $s) {
+            return '';
+        }
+
+        if ($K && null === self::$KD) {
+            self::$KD = self::getData('compatibilityDecomposition');
+        }
+
+        if (null === self::$D) {
+            self::$D = self::getData('canonicalDecomposition');
+            self::$cC = self::getData('combiningClass');
+        }
+
+        if ($C) {
+            if (null === self::$C) {
+                self::$C = self::getData('canonicalComposition');
+            }
+
+            return self::recompose(self::decompose($s, $K));
+        } else {
+            return self::decompose($s, $K);
+        }
+    }
+
+    private static function recompose($s)
+    {
+        $ASCII = self::$ASCII;
+        $compMap = self::$C;
+        $combClass = self::$cC;
+        $ulenMask = self::$ulenMask;
+
+        $result = $tail = '';
+
+        $i = $s[0] < "\x80" ? 1 : $ulenMask[$s[0] & "\xF0"];
+        $len = strlen($s);
+
+        $lastUchr = substr($s, 0, $i);
+        $lastUcls = isset($combClass[$lastUchr]) ? 256 : 0;
+
+        while ($i < $len) {
+            if ($s[$i] < "\x80") {
+                // ASCII chars
+
+                if ($tail) {
+                    $lastUchr .= $tail;
+                    $tail = '';
+                }
+
+                if ($j = strspn($s, $ASCII, $i + 1)) {
+                    $lastUchr .= substr($s, $i, $j);
+                    $i += $j;
+                }
+
+                $result .= $lastUchr;
+                $lastUchr = $s[$i];
+                ++$i;
+            } else {
```

Comment:
@nicolas-grekas Hi I'm just a random php developer that uses Symfony so feel free to disregard my comment.
You can remove this 'else' by adding a 'continue' at the end of the sibling 'if' above; visually makes it less nested.
Same for the 'if' on line 130.